### DISCUSSION TITLE
Realtime - Client ID

### DISCUSSION THREAD

**Discussion by nayakanirudh** [pocketbase/pocketbase#4408] [Q&A] [answered]
Sorry for the basic question, I must be missing something very simple, but I can't figure it out

The docs say a get request on api/realtime will reply with the client ID. I can then set authorization, collection etc. in a subsequent post request with the client ID. Works perfectly in Postman. But when I create an EventSource in JS like below, I can't find the client ID in the response...

```js
const evtSource = new EventSource('https://pb.xyz/api/realtime/')
evtSource.onmessage = (event) => {
console.log(event.data)
}
```

**Comment by ganigeorgiev** [comment, marked as answer]
I haven't tested it but I think the `onmessage` function fires only for "anonymous"/unnamed events. For named events (as it is the case with the PocketBase `PB_CONNECT`) you can try attaching a listener:

```js
evtSource.addEventListener("PB_CONNECT", (e) => {
  console.log(e);
})
```

Is there any specific reason why you can't use the JS SDK? Under the hood it also utilizes whatever global `EventSource` implementation is available.

**Reply by ganigeorgiev** [reply]
From the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/EventSource):

> If there is an event field in the incoming message, the triggered event is the same as the event field value. If no event field is present, then a generic [message](https://developer.mozilla.org/en-US/docs/Web/API/EventSource/message_event) event is fired.
> ...
> To listen to named events, you'll require a listener for each type of event sent.

**Reply by nayakanirudh** [reply]
Of course! I didn't realize that PB_CONNECT itself was an event. Thank you, I got the client ID now. I cannot use the SDK because it doesn't play well with alpine.js; x-data variables don't update via the subscribe(). I am not sure whether this manual approach works either, I will try it now

**Reply by nayakanirudh** [reply]
A manual connection works fine as I can call the event listener within alpine's context. For anybody's future reference using pocketbase with alpine.js, below is an example:

@ganigeorgiev the subscribe() does not reactively update the `verified` variable

```html

<body x-data="app_name" x-cloak>

    <h2 class="bg-gray-300" x-text="verified"></h2>

</body>

<script src="../static/scripts/pocketbase.umd.js"></script>
<script defer src="https://cdn.jsdelivr.net/npm/alpinejs@3.13.5/dist/cdn.min.js"></script>

<script>
    document.addEventListener('alpine:init', () => {
        Alpine.data('app_name', () => ({
            pb: null,
            client_id: '',
            verified: '',
            async init() {

                this.pb = new PocketBase('https://pb.xyz')

                const evtSource = new EventSource('https://pb.xyz/api/realtime/')
                evtSource.addEventListener("PB_CONNECT", (e) => {
                    const data = JSON.parse(e.data)
                    this.client_id = data.clientId
                    console.log('Connected to server with client_id: ' + this.client_id)
                    fetch('https://pb.xyz/api/realtime/', {
                        method: 'POST',
                        headers: {
                            'Content-Type': 'application/json',
                            'Authorization': this.pb.authStore.token,
                        },
                        body: JSON.stringify({
                            "clientId": this.client_id,
                            "subscriptions": ["users/sw4triuc9mkgm3x"]
                        })
                    })
                    .then(response => {
                        console.log('Status:', response.status);
                        return response.json();
                    })
                    .then(data => console.log(data))
                    .catch(error => console.error('Error:', error));
                })



               // this works
                evtSource.addEventListener("users/sw4triuc9mkgm3x", (e) => {
                    console.log('User updated (manual): ', e.data);
                    this.verified = e.data;
                })

               // this does not work
                this.pb.collection('users').subscribe(this.pb.authStore.model.id, function (e) {
                    if (e.action === 'update') {
                        console.log('User updated (subscribe): ', e.record.verified);
                        this.verified = e.record.verified;
                    }
                }});

            },
        }))
    })
</script>

</html>
```

**Reply by ganigeorgiev** [reply]
@EliteEPM Both examples should work correctly. The difference is that in the PocketBase one you are not using an arrow function and `this` will be a different object. Or in other replace `function (e)` with `(e) =>`

**Reply by nayakanirudh** [reply]
Thank you! JS beat me once again
